coreMacros/src/main/scala/Chisel/internal/sourceinfo/SourceInfoTransform.scala

```diff
@@ -6,7 +6,7 @@
 package Chisel.internal.sourceinfo
 
 import scala.language.experimental.macros
-import scala.reflect.macros.blackbox.Context
+import scala.reflect.macros.whitebox.Context
```

Comment:
Are you suggesting making macros blackbox where they don't need this funny override return type like behavior? I think that might make things inconsistent, and it's not immediately clear from method signatures which transforms need whitebox (we aren't consistent with which do_ methods are final, and anything can be type-refined). While sealed classes can limit the impact to changes in Chisel core, it still might end up as a game of whackamole. Though it can also limit the uses of whitebox, making a eventual transition to blackbox easier.